modules/clients/src/test/java/org/apache/ignite/internal/processors/rest/JettyRestProcessorAbstractSelfTest.java

```diff
@@ -1636,6 +1644,184 @@ public void testExe() throws Exception {
         assertResponseContainsError(ret, "Failed to find mandatory parameter in request: id");
     }
 
+    /**
+     * Tests work of VisorCacheGetValueTask via {@link VisorGatewayTask}.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testVisorGetCacheValueTask() throws Exception {
+        ClusterNode locNode = grid(1).localNode();
+
+        Person p = new Person(100, "John", "Doe", 300);
+
+        // Check integer key.
+        jcache().put(1, p);
+
+        String ret = content(new VisorGatewayArgument(VisorCacheGetValueTask.class)
+            .setNode(locNode)
+            .setTaskArgument(VisorCacheGetValueTaskArg.class)
+            .addArguments(DEFAULT_CACHE_NAME, VisorDataType.INT, "1"));
+
+        info("VisorCacheGetValueTask result for Integer key: " + ret);
+
+        JsonNode res = jsonTaskResult(ret);
+
+        String resStr = res.get("result").toString();
+
+        assertTrue(resStr.contains("\"id\":" + p.getId()));
+        assertTrue(resStr.contains("\"orgId\":" + p.getOrganizationId()));
+        assertTrue(resStr.contains("\"firstName\":\"" + p.getFirstName() + "\""));
+        assertTrue(resStr.contains("\"lastName\":\"" + p.getLastName() + "\""));
+        assertTrue(resStr.contains("\"salary\":" + p.getSalary()));
+
+        // Check UUID key.
+        UUID uuidKey = UUID.randomUUID();
+
+        jcache().put(uuidKey, 2);
+
+        ret = content(new VisorGatewayArgument(VisorCacheGetValueTask.class)
+            .setNode(locNode)
+            .setTaskArgument(VisorCacheGetValueTaskArg.class)
+            .addArguments(DEFAULT_CACHE_NAME, VisorDataType.UUID, uuidKey.toString()));
+
+        info("VisorCacheGetValueTask result for UUID key: " + ret);
+
+        res = jsonTaskResult(ret);
+
+        resStr = res.get("result").toString();
+
+        assertTrue("2".equals(resStr));
+
+        // Check Timestamp key.
+        long timestamp = System.currentTimeMillis();
+
+        jcache().put(new Timestamp(timestamp), 3);
+
+        ret = content(new VisorGatewayArgument(VisorCacheGetValueTask.class)
+            .setNode(locNode)
+            .setTaskArgument(VisorCacheGetValueTaskArg.class)
+            .addArguments(DEFAULT_CACHE_NAME, VisorDataType.TIMESTAMP, timestamp));
+
+        info("VisorCacheGetValueTask result for Timestamp key: " + ret);
+
+        res = jsonTaskResult(ret);
+
+        resStr = res.get("result").toString();
+
+        assertTrue("3".equals(resStr));
+
+        // Check Date key.
```

Comment:
Do we need this?

Reply:
It can be uncommented to reproduce bug with transaction lock.